File: PersonalNotesAPI/PersonalNotesAPI/Models/Note.cs
using PersonalNotesAPI.Requirement;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace PersonalNotesAPI.Models
{
    public class Note
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [StringLength(50)]
        public string Title { get; set; }
        [Required]
        public string Description { get; set; }
        public bool Finished { get; set; }
        [RequirementByNotebookID]
        public int NotebookId { get; set; }
        [Required]
        [StringLength(50)]
        public string CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        [StringLength(50)]
        public string UpdatedBy { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public bool Deleted { get; set; }
        [Timestamp]
        public byte[] Timestamp { get; set; }

        [ForeignKey("NotebookId")]
        public virtual Notebook Notebook { get; set; }
    }
}

File: Assignment4/src/app/models/note.ts
export interface INote {
  id: number;

  title: string;

  description: string;

  finished: boolean;

  notebookId: number;

  createdBy: string;

  createdOn: Date | string;

  updatedBy: string;

  updatedOn: Date | string | null;

  deleted: boolean;

  timestamp: string;
}

export class Note implements INote {
  title: string;
  id: number; 
  description: string;
  finished: boolean;
  notebookId: number;
  createdBy: string;
  createdOn: string | Date;
  updatedBy: string;
  updatedOn: string | Date;
  deleted: boolean;
  timestamp: string;
}
